src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs

```diff
@@ -40,14 +40,20 @@ public WebHostBuilder()
             _configureServicesDelegates = new List<Action<IServiceCollection>>();
             _configureLoggingDelegates = new List<Action<ILoggerFactory>>();
 
+            _config = new ConfigurationBuilder()
+                .AddEnvironmentVariables(prefix: "ASPNETCORE_")
+                .Build();
+
             // This may end up storing null, but that's indistinguishable from not adding it.
             UseSetting(WebHostDefaults.EnvironmentKey, Environment.GetEnvironmentVariable("ASPNETCORE_ENVIRONMENT")
```

Comment:
It wouldn't matter because it would just set those configurations again. It's idempotent so it's not wrong but I do think your suggestion looks a little cleaner. Though in this case we need to first create an InMemoryCollection, add the old keys, create a new configuration builder with environment variables and add call `UseConfiguration` which is a lot of fluff. *shrug*